### PR TITLE
adding fees and rewards section

### PR DESCRIPTION
**Pull request by CaisManai**
adding fees and rewards section

### CODE REVIEW

File: fees-rewards.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Fees and Rewards
+
+Obscuro sets out a simple structure for fees and rewards. End users pay fees based on their usage which are placed into a pool from which rewards for nodes are paid out.
+
+The structure delivers a steady and predictable income for node operators over the long term and a predictable fee for users proportional to the actual costs of running the network and does not rely on bidding wars or vary according to activity.
+
+The structure also doesn&#39;t punish Aggregators that miss out on a publishing opportunity with slashing. Instead, the loss of reward should be incentive enough.
```
Comment (tudor-malene) [inline fees-rewards.md]: does this doesn&#39;t render ok?

File: fees-rewards.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+## Cost of L2 nodes
+
+Operators run the network and incur operational costs by running and maintaining nodes.
+
+An &quot;expected monthly node operational costs&quot; variable is set through the governance protocol, which is designed to reflect how much a node operator would pay for hardware, hosting etc, in a month. This is analogous to the proof of work difficulty found in other blockchains. This effects demand for running a node.
+
+Aggregator nodes incur cost from submitting transactions to L1, through the following:
+
+- Storing rollup data on Ethereum.
+- The overhead in participating in publishing rollups outside of the storage cost, e.g. signing transactions.
```
Comment (tudor-malene) [inline fees-rewards.md]: - the overhead of processing the Ethereum transaction that contains the rollup.

File: fees-rewards.md (hunk 1 above)
Comment (GavT) [inline fees-rewards.md]: We using more formal phrases in the WP therefore "doesn't" should be "does not"

File: fees-rewards.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+The structure delivers a steady and predictable income for node operators over the long term and a predictable fee for users proportional to the actual costs of running the network and does not rely on bidding wars or vary according to activity.
+
+The structure also doesn&#39;t punish Aggregators that miss out on a publishing opportunity with slashing. Instead, the loss of reward should be incentive enough.
+
+## Cost of L2 nodes
+
+Operators run the network and incur operational costs by running and maintaining nodes.
+
+An &quot;expected monthly node operational costs&quot; variable is set through the governance protocol, which is designed to reflect how much a node operator would pay for hardware, hosting etc, in a month. This is analogous to the proof of work difficulty found in other blockchains. This effects demand for running a node.
```
Comment (GavT) [inline fees-rewards.md]: affects not effects

File: fees-rewards.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+1. A proportion of the monthly operational cost for all nodes. The proportion is calculated based on the computation of the current transaction as a proportion of the moving average of all L2 transactions over the month. Represented as:
+
+![equation_1.png](images/equation_1.png)
+
+![equation_2.png](images/equation_2.png)
+
+2. The cost incurred by nodes for L1 storage, represented as:
+
+![equation_2.png](images/equation_3.png)
```
Comment (tudor-malene) [inline fees-rewards.md]: shouldn't we give these equations meaningful names?
Don't know how this renders

File: fees-rewards.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+Operators run the network and incur operational costs by running and maintaining nodes.
+
+An &quot;expected monthly node operational costs&quot; variable is set through the governance protocol, which is designed to reflect how much a node operator would pay for hardware, hosting etc, in a month. This is analogous to the proof of work difficulty found in other blockchains. This effects demand for running a node.
+
+Aggregator nodes incur cost from submitting transactions to L1, through the following:
+
+- Storing rollup data on Ethereum.
+- The overhead in participating in publishing rollups outside of the storage cost, e.g. signing transactions.
+
+## Fee per transaction
```
Comment (GavT) [inline fees-rewards.md]: Title needs to be capitalised

File: fees-rewards.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+![equation_2.png](images/equation_3.png)
+
+3. A proportion of the additional overhead cost incurred by nodes for processing rollups is as follows:
+
+The idea behind this is:
+
+- At the point of transaction, it&#39;s impossible to know what the true cost of the future rollup is
+- Users need to pay their fair share of what this future rollup cost will be
+- The best indicator of what the future rollup cost might be is a moving average of the previous n blocks
```
Comment (tudor-malene) [inline fees-rewards.md]: n rollups

File: fees-rewards.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+- Where n is high, the cost is averaged out amongst more blocks and therefore less volatile and vice versa. High n adds more risk to Aggregators
```
Comment (tudor-malene) [inline fees-rewards.md]: if the averaging is performed over a larger number of block....

File: fees-rewards.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+Aggregator nodes incur cost from submitting transactions to L1, through the following:
+
+- Storing rollup data on Ethereum.
+- The overhead in participating in publishing rollups outside of the storage cost, e.g. signing transactions.
+
+## Fee per transaction
+
+The total fee a user must pay has three components.
+
+1. A proportion of the monthly operational cost for all nodes. The proportion is calculated based on the computation of the current transaction as a proportion of the moving average of all L2 transactions over the month. Represented as:
```
Comment (GavT) [inline fees-rewards.md]: Proportion used twice in one sentence. Should it read "This is calculated based on the computation of the current transaction as a proportion of the..."?

File: fees-rewards.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+![equation_1.png](images/equation_1.png)
+
+![equation_2.png](images/equation_2.png)
+
+2. The cost incurred by nodes for L1 storage, represented as:
+
+![equation_2.png](images/equation_3.png)
+
+3. A proportion of the additional overhead cost incurred by nodes for processing rollups is as follows:
```
Comment (GavT) [inline fees-rewards.md:34]: This sentence then goes to the another sentence ending in a semi-colon. Needs to be tidied up a bit.

File: fees-rewards.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+The idea behind this is:
+
+- At the point of transaction, it&#39;s impossible to know what the true cost of the future rollup is
```
Comment (GavT) [inline fees-rewards.md]: It it

File: fees-rewards.md (hunk 6 above)
Comment (GavT) [inline fees-rewards.md]: The best indicator or an indicator? Statisticians might argue this is not the best

File: fees-rewards.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+- Users need to pay their fair share of what this future rollup cost will be
+- The best indicator of what the future rollup cost might be is a moving average of the previous n blocks
+- Where n is high, the cost is averaged out amongst more blocks and therefore less volatile and vice versa. High n adds more risk to Aggregators
+- L2 gas can be computed as Obscuro has the same metering mechanism as L1. As a result of this metering, we can work out the gas consumed and directly map this to the L1 moving average gas cost
```
Comment (tudor-malene) [inline fees-rewards.md]: The computation required by Obscuro nodes to run a transaction can be estimated by using a metering mechanism similar to the one found in Ethereum.  
Using this information we can calculate what is the share of each transaction from the total computation in an average block.
The basis to calculate the L1 fee is the cost to publish an average rollup to L1.
Each transaction will pay a share of that cost proportional to its computation

File: fees-rewards.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+This gives the expected fee per transaction as:
+
+![equation_4.png](images/equation_5.png)
+
+##
+
+## Rewards for L2 nodes
+
+The Obscuro protocol rewards both Validators and Aggregators that actively monitor the network via a lottery. The lottery randomly rewards active nodes each round with an OBX reward for participating. Nodes prove they&#39;re alive by claiming their reward within the subsequent few blocks; else, the reward is burned. The reward in each rollup is calculated to cover the operational costs of active nodes.
```
Comment (tudor-malene) [inline fees-rewards.md]: This paragraph uses "active" a lot. 
I think it could be shortened a bit

File: fees-rewards.md (hunk 6 above)
Comment (GavT) [reply fees-rewards.md]: "Previous n blocks" or "previous n rollups"?

File: fees-rewards.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+Aggregator nodes have an additional reward mechanism that pays out rewards in OBX when they successfully win the right to validate and publish a rollup. This reward covers at least the L1 gas costs incurred.
```
Comment (tudor-malene) [inline fees-rewards.md]: Aggregator nodes have an additional reward mechanism that pays out rewards in OBX after they successfully publish a rollup based on the rules of the POBI protocol. This reward covers at least the L1 gas costs incurred and will not be greater than double the L1 gas cost.

File: fees-rewards.md (hunk 7 above)
Comment (GavT) [reply fees-rewards.md]: I don't understand the rationale for sentence on risk. Can you explain it in more detail?

File: fees-rewards.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+The lottery pay out amount per rollup round is represented as:
+
+![active_node_reward_equation_6.png](images/equation_6.png)
+
+## Bootstrapping the network
+
+##
```
Comment (tudor-malene) [inline fees-rewards.md]: empty heading

File: fees-rewards.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+When the network first spins up, it is prudent to expect low transaction volume. Low transaction volume creates a situation where:
```
Comment (tudor-malene) [inline fees-rewards.md]: When the network first spins up, it is prudent to expect low transaction volume, which can create a situation where:

File: fees-rewards.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+1. Nodes can publish continually at a high cadence and absorb what might amount to a loss-making operation.
```
Comment (tudor-malene) [inline fees-rewards.md]: Aggregators can..

File: fees-rewards.md (hunk 12 above)
Comment (GavT) [reply fees-rewards.md]: "they're" becomes "they are"

Comment (GavT) [reply fees-rewards.md]: Stray semi colon?

File: fees-rewards.md (hunk 2 above)
Comment (tudor-malene) [reply fees-rewards.md]: this doesn't look resolved

File: fees-rewards.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+The lottery pay out (active node reward) amount per rollup round is represented as:
+
+- The lottery reward in each rollup is calculated to cover the operational costs of active nodes.
+
+![equation_5.png](images/equation_6.png)
+
+## Bootstrapping the network
+
+##
```
Comment (GavT) [inline fees-rewards.md]: Stray heading?

File: fees-rewards.md (hunk 11 above)
Comment (CaisManai) [reply fees-rewards.md]: Are these points to add in addition or replace some of these existing points?

File: fees-rewards.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+When the network first spins up, it&#39;s prudent to expect low transaction volume. Low transaction volume creates a situation where:
```
Comment (GavT) [inline fees-rewards.md]: "it's" becomes "it is"

File: fees-rewards.md (hunk 2 above)
Comment (CaisManai) [reply fees-rewards.md]: Try refreshing, it's resolved

File: fees-rewards.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+
+1. Nodes can publish continually at a high cadence and absorb what might amount to a loss-making operation.
+2. Nodes can throttle the rate at which rollups are published, leading to a poorer user experience.
+3. Alternatively, the network can subsidize early transactions allowing both groups to succeed.
```
Comment (GavT) [inline fees-rewards.md]: English spelling so "subsidize" becomes "subsidise"

File: fees-rewards.md [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+![storage_equation.png](images/equation_3.png)
+
+3. A proportion of the additional overhead cost incurred by nodes for processing rollups is as follows:
+
+The idea behind this is:
+
+- At the point of transaction, it is impossible to know what the true cost of the future rollup is
+- Users need to pay their fair share of what this future rollup cost will be
+- The best indicator of what the future rollup cost might be is a moving average of the previous n rollups
+- Where n is high, the averaging is performed over a larger number of rollups and therefore less volatile and vice versa. High n adds more risk to Aggregators
```
Comment (tudor-malene) [inline fees-rewards.md]: Let's remove : High n adds more risk to Aggregators
because it's unclear how it works

File: fees-rewards.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+
+##
+
+When the network first spins up, it&#39;s prudent to expect low transaction volume. Low transaction volume creates a situation where:
+
+1. Nodes can publish continually at a high cadence and absorb what might amount to a loss-making operation.
+2. Nodes can throttle the rate at which rollups are published, leading to a poorer user experience.
+3. Alternatively, the network can subsidize early transactions allowing both groups to succeed.
+
+While Obscuro optimizes for end-users, early nodes should also be profitable. As such, Obscuro follows the path of option 3
```
Comment (GavT) [inline fees-rewards.md]: English spelling so "optimizes" becomes "optimises"

File: fees-rewards.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+
+- In the young days of the network, fees will be subsidized by newly minted tokens to ensure early users do not face excessive fees. The price of a transaction is fixed at $1 in OBX until OBX establishes a stable market.
```
Comment (GavT) [inline fees-rewards.md]: "early days" reads better than "young days"

Comment (GavT) [reply fees-rewards.md]: Subsidised (English spelling)

Comment (GavT) [reply fees-rewards.md]: "until OBX establishes a stable market." is a sensitive statement. Perhaps soften it as "until the Obscuro network becomes better established". or "until the Obscuro network becomes more prevalent."

File: fees-rewards.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+- A period of subsidizing new nodes that reduces over time until transaction volume picks up and market forces take over.
```
Comment (GavT) [inline fees-rewards.md]: Remove "and market forces take over."

Comment (GavT) [reply fees-rewards.md]: Actually, should these 2 bullets still be here? They appear to be duplicates of the 3 rules above.

**Review by GavT — CHANGES_REQUESTED** [review]

File: fees-rewards.md (hunk 3 above)
Comment (CaisManai) [reply fees-rewards.md]: nice spot

File: fees-rewards.md (hunk 8 above)
Comment (CaisManai) [reply fees-rewards.md]: definitely

File: fees-rewards.md (hunk 23 above)
Comment (CaisManai) [reply fees-rewards.md]: Yes, I think they need to be there as they explain how we'll implement #3

File: fees-rewards.md [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+
+When the network first spins up, it is prudent to expect low transaction volume, which can create a situation where:
+
+1. Aggregators can publish continually at a high cadence and absorb what might amount to a loss-making operation.
+2. Aggregators can throttle the rate at which rollups are published, leading to a poorer user experience.
+3. Alternatively, the network can subsidise early transactions allowing both groups to succeed.
+
+While Obscuro optimises for end-users, early nodes should also be profitable. As such, Obscuro follows the path of option 3
+
+- In the early days of the network, fees will be subsidised by newly minted tokens to ensure early users do not face excessive fees. The price of a transaction is fixed at $1 in OBX until the Obscuro network is better established.
```
Comment (GavT) [inline fees-rewards.md]: Why are these bullets rather than a single paragraph?

File: fees-rewards.md [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+
+## Fee Per Transaction
+
+The total fee a user must pay has three components.
+
+1. A proportion of the monthly operational cost for all nodes. This is calculated based on the computation of the current transaction as a proportion of the moving average of all L2 transactions over the month. Represented as:
+
+![proportion_equation.png](images/equation_1.png)
+
+![operation_equation.png](images/equation_2.png)
```
Comment (GavT) [inline fees-rewards.md]: "no of nodes" should be "no. of nodes"

File: fees-rewards.md [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+
+## Rewards For L2 nodes
+
+The Obscuro protocol rewards both Validators and Aggregators that monitor the network via a lottery. The lottery randomly rewards active nodes each round with an OBX reward for participating. Nodes prove they are alive by claiming their reward within the subsequent few blocks; else, the reward is burned. The reward in each rollup is calculated to cover the operational costs of these nodes.
+
+Aggregator nodes have an additional reward mechanism that pays out rewards in OBX when successfully publish a rollup based on the rules of the POBI protocol. This reward covers at least the L1 gas costs incurred and will not be greater than double the L1 gas cost.
+
+The lottery pay out amount per rollup round is represented as:
+
+![active_node_reward_equation_6.png](images/equation_6.png)
```
Comment (GavT) [inline fees-rewards.md]: "no of nodes" becomes "no. of nodes"

File: fees-rewards.md [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+- The computation required by Obscuro nodes to run a transaction is estimated using a metering mechanism similar to the one found in Ethereum.
+- Using this information, Obscuro calculates the share of each transaction from the total computation in an average block.
+- The basis for calculating the L1 fee is the cost to publish an average rollup to L1.
+- Each transaction pays a share of that cost proportional to its computation.
+
+![overhead_equation_4.png](images/equation_4.png)
+
+This gives the expected fee per transaction as:
+
+![total_fee_equation_5.png](images/equation_5.png)
```
Comment (GavT) [inline fees-rewards.md]: There is some inconsistency on the use of capitalisation in the equations

**Review by GavT — CHANGES_REQUESTED** [review]

**Pull request merged**